src/cli.ts

```diff
@@ -0,0 +1,294 @@
+        ' values.  Command-line options flags supercede values in file ' +
+        ' where they differ.  If file is missing, Tattoo will ignore.'
+  },
+  {
+    name: 'github-token',
+    alias: 'g',
+    type: String,
+    description: 'Provide github token via command-line flag instead of ' +
+        '"github-token" file.'
+  },
+  {
+    name: 'latest-release',
+    alias: 'l',
+    type: Boolean,
+    defaultValue: false,
+    description: 'Set to update repos to the latest release when possible.'
+  },
+  {
+    name: 'verbose',
+    alias: 'v',
+    type: Boolean,
+    defaultValue: false,
+    description: 'Set to print output from failed tests.'
+  },
+  {
+    name: 'wct-flags',
+    alias: 'w',
+    multiple: true,
+    type: String,
+    defaultValue: ['--local chrome'],
+    description: 'Set to specify flags passed to wct.'
+  },
+  {
+    name: 'workspace-dir',
+    alias: 'd',
+    type: String,
+    description:
+        'Override the default path "repos" where the repositories will be ' +
+        ' cloned and web-components-tester will run.'
+  },
+  {
+    name: 'help',
+    alias: 'h',
+    type: Boolean,
+    description: 'Print this usage example.'
+  }
+];
+
+/**
+ * Checks for github-token in the RunnerOptions and if not specified, will look
+ * in the github-token file in working folder.  If that doesn't exist either,
+ * we message to the user that we need a token and exit the process.
+ */
+export function ensureGitHubToken(options: CliOptions) {
+  // TODO(usergenic): Maybe support GITHUB_TOKEN as an environment variable,
+  // since this would be a better solution for Travis deployments etc.
+  if (!options['github-token']) {
+    try {
+      options['github-token'] = fs.readFileSync('github-token', 'utf8').trim();
+    } catch (e) {
+      console.error(`
+You need to create a github token and place it in a file named 'github-token'.
+The token only needs the 'public repos' permission.
+
+Generate a token here:   https://github.com/settings/tokens
+    `);
+      process.exit(1);
+    }
+  }
+}
+
+/**
+ * Reads and parses the json config file content if the filename is specified
+ * and the file exists.  If it is not specified or does not exist, returns empty
+ * options object.  Will throw on malformed file only.
+ */
+export function loadConfigFileOptions(options: CliOptions): ConfigFileOptions {
+  if (!options['config-file']) {
+    return {};
+  }
+
+  const configFile = options['config-file'];
+
+  try {
+    if (fs.lstatSync(configFile)) {
+      // TODO(usergenic): Test for file presence and provide proper error
+      // message.
+      return JSON.parse(fs.readFileSync(configFile, 'utf8'));
+    }
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      return {};
+    }
+    throw(err);
+  }
+
+  return {};
+}
+
+/**
+ * Loads the config file specified in the options as 'config-file' and merges
+ * the values into the provided options as appropriate.
+ */
+export function mergeConfigFileOptions(
```

Comment:
A potentially useful abstraction: a command-line-args compatible resource-file package that supports a lot of the standard .somethingrc/something.json style behaviors, including support for directory walkback etc.  I'll look out for one.